compactor/src/utils.rs

```diff
@@ -199,17 +200,33 @@ pub(crate) fn compute_split_time(
     let mut min = min_time;
     loop {
         let split_time = min + ((max_time - min_time) as f64 * percentage).ceil() as i64;
-        if split_time < max_time {
-            split_times.push(split_time);
-            min = split_time;
-        } else {
+
+        if split_time >= max_time {
             break;
+        } else if time_range_present(chunk_times.clone(), min, split_time) {
```

Comment:
Why is this `clone` needed? While it *shouldn't* be too expensive to clone a bunch of `i64`s, I don't think it's *needed*. I think this can be a reference:

```suggestion
        } else if time_range_present(&chunk_times, min, split_time) {
```

and then more changes coming below...

Reply:
Thanks!  I tried to get rid of the clone using an Arc, but didn't I didn't get that to compile.